samples/RaspberryPiInstructions.md

```diff
@@ -1,70 +1,129 @@
-```
-sudo apt-get update
-sudo apt-get install curl libunwind8 gettext apt-transport-https
+  Note: The latest download links are available at the [.NET Core Download page](https://dotnet.microsoft.com/download/archives).
+2. Unpack .NET Core to a global location:
+  * `sudo mkdir -p /usr/share/dotnet`
+  * `sudo tar -zxf dotnet.tar.gz -C /usr/share/dotnet`
+3. Symbolically link dotnet into a path location:
+  * `sudo ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet`
+
+You should be able to run .NET Core at this point:
+
+```console
+dotnet
 ```
 
-* Copy your app, i.e. whole `publish` directory mentioned above, to the Raspberry Pi and execute run `./helloworld` to see `Hello World!` from .NET Core running on your Pi! (make sure you `chmod 755 ./helloworld`)
+## Installing the .NET Core SDK on Raspberry Pi on Win10 IoT Core
+
+Install [Windows 10 IoT Core](https://docs.microsoft.com/en-us/windows/iot-core/getstarted) on your Pi.
+
+Install the [.NET Core 3.0 SDK for Windows ARM32](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+
+## Creating and Running an Application on the Pi
+
+It is easy to run an application on the Pi. The following commands are for Linux, but switching the casing will enable the same commands to work on Windows.
+
+Console App:
+
+1. `dotnet new console -o app` // Generate app from template
+2. `cd app`
+3. `dotnet run` // Build and run app
+4. `./bin/Debug/netcoreapp3.0/app` // Run app directly (once built)
+
+Web App:
+
+1. `dotnet new webapp -o webapp` // Generate app from template
+2. `cd webapp`
+3. `dotnet run` // Build and run app -- will use `localhost`
+4. `./bin/Debug/netcoreapp3.0/webapp --urls "http://*:5100"` //Run app, exposed on port 5100
+
+Note: `dotnet run` will use the `localhost` loopback by default. To view the site on an external port, [configure ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/host/web-host?view=aspnetcore-2.1#server-urls) to do so with `--urls "http://*:5100"` argument (with `dotnet run` or direct exectuable launch, as seen in step 4 above) and then view on the appropriate IP address and port, such as `http://192.168.1.75:5100`.
+
+## Build on X64 and Publish to Pi
+
+You can build and publish app on another machine (likely x64) and publish to the Pi. We will use a webapp for this example and first test on the local machine (desktop or laptop).
+
+1. `dotnet new webapp -o webapp` // Generate app from template
+2. `cd webapp`
+3. `dotnet run` // Build and run app -- will use `localhost:5000`
+4. `\bin\Debug\netcoreapp3.0\webapp.exe` // Run app directly (once built)
+
+Now, we will copy the app to the Pi with scp (other tools work similarly, like Putty). There are a few different options.
+
+Assume matching .NET Core runtime is installed on Pi:
 
-### Win10 IoT Core
+1. `scp -r bin\Debug\netcoreapp3.0\ pi@192.168.1.75:~/webapp` // copy app to Pi
```

Comment:
Why are we copying over the build output? What we should be doing here is to do a portable publish (FDD) and copying the published output to the deployment machine.

Reply:
In 3.0, you are supposed to be able to copy build output. Does that not work for asp.net?

Reply:
Ah, I see. I was under the impression you'd want this to work in 2.2 as well and I was thinking along those lines.

Reply:
Good point. It's tough to make the docs work for 2.2 vs 3.0, windows vs linux and 32bit vs 64bit. I will endeavour to make that work.